src/values/instruction_value.rs

```diff
@@ -675,6 +675,18 @@ impl<'ctx> InstructionValue<'ctx> {
 
         Ok(())
     }
+
+    /// Return this `InstructionValue` as a `PhiValue`.
+    /// Panics if the instruction is not a Phi instruction.
+    pub fn as_phi_value(&self) -> PhiValue<'ctx> {
+        if self.get_opcode() == InstructionOpcode::Phi {
+            unsafe {
+                PhiValue::new(self.as_value_ref())
+            }
+        } else {
+            panic!("Found {:?} but expected the PhiValue variant", self)
```

Comment:
This should be a None rather than panicing

Reply:
Or even `TryFrom` impl

Reply:
This is switched to `TryFrom` now, can you give it another look @TheDan64? Thanks!